apps/src/code-studio/header.js

```diff
@@ -168,10 +170,44 @@ function setupReduxSubscribers(store) {
 setupReduxSubscribers(getStore());
 
 function setUpGlobalData(store) {
-  store.dispatch(asyncLoadUserData());
+  fetch('/api/v1/users/current', {
+    credentials: 'same-origin'
+  })
+    .then(response => response.json())
+    .then(data => {
+      store.dispatch(setUserSignedIn(data.is_signed_in));
+      if (data.is_signed_in) {
+        store.dispatch(setInitialData(data));
+        ensureHeaderSigninState(true, data.short_name);
+      } else {
+        ensureHeaderSigninState(false);
+      }
+    })
+    .catch(err => {
+      console.log(err);
+    });
 }
 setUpGlobalData(getStore());
 
+// Some of our cached pages can become cached by the browser with the
+// wrong sign-in state. This is a temporary patch to ensure that the header
+// displays the correct sign-in state for the user.
+function ensureHeaderSigninState(isSignedIn, shortName) {
+  const userMenu = document.querySelector('#header_user_menu');
+  const signinButton = document.querySelector('#signin_button');
+
+  if (isSignedIn && userMenu.style.display === 'none') {
+    userMenu.style.display = 'block';
+    signinButton.style.display = 'none';
+
+    const displayName = document.querySelector('#header_display_name');
+    displayName.textContent = shortName;
+  } else if (!isSignedIn && signinButton.style.display === 'none') {
```

Comment:
It strikes me that this would be the default- this is a NIT but it might be organizes as just a default 'let' for those two vars with a single 'if' instead.

Reply:
I'm not sure I understand what you mean by default 'let' for those two vars. I may follow-up with you on this and get these changes in separately!